src/main/webapp/scripts/forum-script.js

```diff
@@ -179,3 +211,37 @@ function collapseReplies(idHandler) {
   $('#' + elementId + ' .expand-button').css('display', 'inline-block');
   $('#' + elementId + ' .collapse-button').css('display', 'none');
 }
+
+/**
+ *  Onclick handler for a user searching to reload forum with search parameter
+ */
+function search() {
+  const search = $('#search-input').val();
+  $('#search-input').val('');
+  const id = -1;
+  const placeholder = $('#forum-placeholder');
+  expandForumWithSearch(placeholder, id, search);
+}
+
+/**
+ *  Returns whether a word of the search is contained with in the text
+ *
+ *  @param {String} text contents of a question
+ *  @param {String} search contents of the user search
+ *
+ *  @return {Boolean} whether there is a word of the search within the text
+ */
+function containsSearch(text, search) {
+  const stopWords = ['a', 'to', 'and', 'how', 'the', 'when', 'what', 'why',
+    'what', 'where', 'or', 'do', 'can', 'use', 'i', 'you', 'my', 'your'];
+  const lowerText = text.toLowerCase();
+  const lowerSearch = search.toLowerCase();
+  let words = lowerSearch.split(' ');
+  words = words.filter( function(word) {
+    return ((!stopWords.includes(word)) && (lowerText.includes(word)));
+  });
+  if (words.length > 0) {
+    return true;
+  }
+  return false;
```

Comment:
nitpick: Simplify lines 243-246 to just `return words.length > 0;`

Reply:
Updated.